persistence/src/test/java/cz/muni/fi/pa165/mushrooms/dao/VisitDaoTest.java

```diff
@@ -0,0 +1,384 @@
+package cz.muni.fi.pa165.mushrooms.dao;
+
+import cz.muni.fi.pa165.mushrooms.entity.Forest;
+import cz.muni.fi.pa165.mushrooms.entity.Mushroom;
+import cz.muni.fi.pa165.mushrooms.entity.MushroomHunter;
+import cz.muni.fi.pa165.mushrooms.entity.Visit;
+import cz.muni.fi.pa165.mushrooms.validation.PersistenceSampleApplicationContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.AbstractTransactionalJUnit4SpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.validation.ConstraintViolationException;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * @author BohdanCvejn
+ */
+
+@ContextConfiguration(classes = PersistenceSampleApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+public class VisitDaoTest extends AbstractTransactionalJUnit4SpringContextTests {
+
+    @Autowired
+    private VisitDao visitDao;
+
+    @PersistenceContext
+    private EntityManager em;
+
+    private Visit visit1;
+    private Visit visit2;
+
+    private Forest forest1;
+    private Forest forest2;
+
+    private MushroomHunter mushroomHunter1;
+    private MushroomHunter mushroomHunter2;
+
+    @Before
+    public void setUp() {
+        forest1 = new Forest();
+        forest1.setName("Dun Morogh");
+        forest1.setDescription("Dun Morogh forest in South Wisconsin");
+
+        forest2 = new Forest();
+        forest2.setName("Marola Xades");
+        forest2.setDescription("Marola Xades forest in South America");
+
+        mushroomHunter1 = new MushroomHunter();
+        mushroomHunter1.setFirstName("Alex");
+        mushroomHunter1.setSurname("Black");
+        mushroomHunter1.setUserNickname("Blacky");
+
+        mushroomHunter2 = new MushroomHunter();
+        mushroomHunter2.setFirstName("Piper");
+        mushroomHunter2.setSurname("White");
+        mushroomHunter2.setUserNickname("Whity");
+
+        visit1 = new Visit();
+        visit1.setHunter(mushroomHunter1);
+        visit1.setForest(forest1);
+        visit1.setDate(LocalDate.of(2017,10,25));
+
+        visit2 = new Visit();
+        visit2.setHunter(mushroomHunter2);
+        visit2.setForest(forest2);
+        visit2.setDate(LocalDate.of(2017,10,30));
+
```

Comment:
Visits should be added to Forest and MushroomHunter instances too (yes we are missing the appropriate methods) - should not be the focus of the test for now, though.